cpp/basix/polyset.cpp

```diff
@@ -680,51 +719,134 @@ void tabulate_polyset_hex_derivs(xt::xtensor<double, 3>& P, std::size_t n,
   const std::size_t m = (n + 1) * (n + 1) * (n + 1);
   const std::size_t md = (nderiv + 1) * (nderiv + 2) * (nderiv + 3) / 6;
 
+  // Indexing for hexahedral basis functions
+  auto hex_idx
+      = [n](std::size_t px, std::size_t py, std::size_t pz) -> std::size_t
+  { return (n + 1) * (n + 1) * px + (n + 1) * py + pz; };
+
   // Compute 1D basis
   const xt::xtensor<double, 1> x0 = xt::col(x, 0);
   const xt::xtensor<double, 1> x1 = xt::col(x, 1);
   const xt::xtensor<double, 1> x2 = xt::col(x, 2);
-  // FIXME: remove this memory assignment
-  xt::xtensor<double, 3> px({static_cast<std::size_t>(nderiv + 1), x0.shape(0),
-                             static_cast<std::size_t>(n + 1)});
-  tabulate_polyset_line_derivs(px, n, nderiv, x0);
-  // FIXME: remove this memory assignment
-  xt::xtensor<double, 3> py({static_cast<std::size_t>(nderiv + 1), x1.shape(0),
-                             static_cast<std::size_t>(n + 1)});
-  tabulate_polyset_line_derivs(py, n, nderiv, x1);
-  // FIXME: remove this memory assignment
-  xt::xtensor<double, 3> pz({static_cast<std::size_t>(nderiv + 1), x2.shape(0),
-                             static_cast<std::size_t>(n + 1)});
-  tabulate_polyset_line_derivs(pz, n, nderiv, x2);
-
-  // Compute basis
+
+  assert(x0.shape(0) > 0);
+  assert(x1.shape(0) > 0);
+  assert(x2.shape(0) > 0);
+  const auto X0 = x0 * 2.0 - 1.0;
```

Comment:
These could be doing allocations. Not sure how lazy the evaluations are, or how to check.

Reply:
I've removed these